### PR TITLE
chore(remix-dev): remove broken param collision check

### PR DESCRIPTION
**Pull request by mcansh**
hit a snag in a demo app as it complained there was a conflict when there wasn't.

```log
⚠️ Route Path Collision: "/:username"

The following routes all define the same URL, only the first one will be used

🟢 routes/_user.$username.tsx
⭕️️ routes/sneakers.$sneakerId.tsx
```

Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

File: packages/remix-dev/__tests__/flat-routes-test.ts [hunk 1]
```diff
@@ -619,6 +619,33 @@ describe("flatRoutes", () => {
     }
   });
 
+  describe("doesn't warn when there's not a route collision", () => {
```
Comment (mcansh) [inline packages/remix-dev/__tests__/flat-routes-test.ts:622]: this test case is here for when param collision is brought back

**Comment by mcansh** [comment]
i think we can skip the changeset here as this was never in a stable release yet

File: packages/remix-dev/__tests__/flat-routes-test.ts [hunk 2]
```diff
@@ -666,7 +692,7 @@ describe("flatRoutes", () => {
       );
     });
 
-    test("same path, different param name", () => {
+    test.skip("same path, different param name", () => {
```
Comment (mcansh) [inline packages/remix-dev/__tests__/flat-routes-test.ts:695]: remove skip once new conflict detection is brought in

File: packages/remix-dev/__tests__/flat-routes-test.ts [hunk 3]
```diff
@@ -619,6 +619,33 @@ describe("flatRoutes", () => {
...
+        APP_DIR,
+        testFiles.map((file) => path.join(APP_DIR, normalizePath(file)))
+      );
+
+      let routes = Object.values(routeManifest);
+
+      expect(routes).toHaveLength(testFiles.length);
+      expect(consoleError).not.toHaveBeenCalledWith(
+        getRouteConflictErrorMessage("/:username", testFiles)
+      );
```
Comment (brophdawg11) [inline packages/remix-dev/__tests__/flat-routes-test.ts:645]: nit: could probably make this a stricter assertion to ensure no errors got logged

```suggestion
      expect(consoleError).not.toHaveBeenCalled();
```

**Review by brophdawg11 — APPROVED** [review]

**Comment by brophdawg11** [comment]
Ah - @mcansh we will need a changeset here since this points to `release-next` - otherwise it won't trigger a new release.  But we can remove the actual line from the final release notes since this never shipped

**Comment by mcansh** [comment]
> Ah - @mcansh we will need a changeset here since this points to `release-next` - otherwise it won't trigger a new release. But we can remove the actual line from the final release notes since this never shipped

👍 this is the fella to remove `.changeset/afraid-tigers-occur.md`

**Pull request merged**